src/components/ImageModal/index.tsx

```diff
@@ -0,0 +1,344 @@
+interface StyledImageContainerProps {
+  translateValue: TranslateValue
+}
+interface StyledImageProps {
+  isFixedHeight: boolean
+}
+interface StyledGradientProps {
+  direction: 'top' | 'bottom'
+}
+
+const RESIZE_HEIGHT = 640
+const IMAGE_GAP_HALF = 6
+const TOUCH_START_END_DIRRERENCE = 30
+
+export const ImageModal = ({
+  onClose,
+  parentElement,
+  images,
+  isOpen = false,
+  name,
+  ...props
+}: ImageModalProps): ReactElement => {
+  const imagesInfo = images.map(({ src, id }) => {
+    const image = new Image()
+    image.src = src
+    const height = image.height
+    const resizeWidth = image.width * (RESIZE_HEIGHT / height)
+
+    return {
+      height,
+      id,
+      resizeWidth,
+      src
+    }
+  })
+  const firstImageWidth = imagesInfo[0].resizeWidth / 2 + IMAGE_GAP_HALF
+  const firstImageId = imagesInfo[0].id
+  const initialTranslateValue = {
+    imageCount: 0,
+    imageWidth: firstImageWidth
+  }
+  const [translateValue, setTranslateValue] = useState<TranslateValue>(
+    initialTranslateValue
+  )
+  const [selectedImageId, setSelectedImageId] = useState<string>(firstImageId)
+  const clientX = useRef<number | null>(null)
+
+  useEffect(() => {
+    setSelectedImageId(firstImageId)
+    setTranslateValue(initialTranslateValue)
+  }, [isOpen])
+
+  const getNextTranslateValue = (selectedId: string): TranslateValue =>
+    [...imagesInfo].reduce(
+      (sumValue, currentImage, idx, images) => {
+        const { imageWidth: prevImageWidth } = sumValue
+        const { id } = currentImage
+        const resizeImageWidth = imagesInfo[idx].resizeWidth
+
+        if (id === selectedId) {
+          // for early return
+          images.splice(1)
+          return {
+            imageCount: idx,
+            imageWidth: prevImageWidth + resizeImageWidth / 2 + IMAGE_GAP_HALF
+          }
+        }
+
+        return {
+          imageCount: idx,
+          imageWidth: prevImageWidth + resizeImageWidth + IMAGE_GAP_HALF
+        }
+      },
+      {
+        imageCount: 0,
+        imageWidth: 0
+      }
+    )
+
+  const handleClickIndicator: MouseEventHandler<HTMLDivElement> = (e): void => {
+    const selectedId = e.currentTarget.dataset.id || ''
+
+    setSelectedImageId(selectedId)
+    setTranslateValue(getNextTranslateValue(selectedId))
+  }
+
+  const hadnleTouchStart: TouchEventHandler = (e): void => {
+    if (clientX.current === null) {
+      clientX.current = e.changedTouches[0].pageX
+    }
+  }
+
+  const handleTouchEnd: TouchEventHandler = (e): void => {
+    const prevTouchX = clientX.current
+    const curTouchX = e.changedTouches[0].pageX
+    const currentImageIdx = translateValue.imageCount
+
+    if (prevTouchX === null) {
+      return
+    }
+
+    if (prevTouchX - curTouchX > TOUCH_START_END_DIRRERENCE) {
+      const nextImageIndex = currentImageIdx + 1
+      const nextImage = imagesInfo[nextImageIndex]
+      const updateImageId = nextImage ? nextImage.id : imagesInfo[0].id
```

Comment:
```suggestion
      const newCurrentImageId = nextImage ? nextImage.id : imagesInfo[0].id
```

`updateImageId`보다 현재 이미지를 나타내는 네이밍이 좋을 것 같습니다!

Reply:
[요기](https://github.com/price-offer/offer-design-system/pull/35/commits/0679b5977c191cc0670edb15a87665fd7f29f4c2)에 반영 완료 ! ☺️